src/sentry/tasks/relay.py

```diff
@@ -254,3 +273,46 @@ def schedule_invalidate_project_config(
     projectconfig_debounce_cache.invalidation.debounce(
         public_key=public_key, project_id=project_id, organization_id=organization_id
     )
+
+
+@instrumented_task(
+    name="sentry.tasks.relay.check_build_project_config",
+    queue="relay_config_check",
+)
+def check_build_project_config(public_key=None, **kwargs):
+    """Temporary task to verify a build task produces a cache entry
+
+    A build task should result in a cache entry being written.  Sometimes it seems this
+    doesn't work.  This task checks for cache entries and logs data if they are missing.  It
+    should be scheduled together with the build task itself.
+
+    This task needs to be scheduled in the future together with the original task.
+    """
+    from sentry.models import ProjectKey
+
+    metrics.incr(
+        "relay.projectconfig_cache.run",
+        tags={"task": "check"},
+        sample_rate=1,
+    )
+    if not public_key:
```

Comment:
nit
```suggestion
    if public_key is None:
```
?

Reply:
The "Pythonic" advise is to not explicitly compare with None but rather rely on "truthness".  E.g. someone could be sending in an empty list, emtpy dict or empty string and this check would still do the right thing.  Admittedly this isn't a substitute for strong typing, but it generally goes rather well with duck-typing things.
There are afaik no other "falsy" values that would be correct here so I feel that this general pythonic style is warranted.